dts: bindings: Drop cat1 from the infineon binding files

sreeramIfx opened the pull request:
Drop cat1 from the binding files to enable reuse by other category devices as well.

Fixes #99174

sreeramIfx committed:
dts: bindings: Drop cat1 from the infineon binding files

Drop cat1 from the binding files to enable reuse by other
category devices as well.

Fixes #99174

Signed-off-by: Sreeram Tatapudi <sreeram.praveen@infineon.com>

maass-hamburg reviewed (dismissed):
these changes need to be mentioned in the migration guide

sreeramIfx committed:
doc: releases: migration: 4.4: Add infineon driver rename update

Adding details regarding the Infineon driver refactoring to drop
cat1 from the driver names and binding files.

Signed-off-by: Sreeram Tatapudi <sreeram.praveen@infineon.com>

jsbatch commented on samples/drivers/uart/async_api/boards/cy8cproto_062_4343w.overlay:
Not really a blocking issue, but header is missing from this file.

jsbatch commented on tests/drivers/uart/uart_async_api/boards/cy8cproto_062_4343w.overlay:
Not really a blocking issue, but header is missing from this file.

jsbatch commented on tests/drivers/uart/uart_async_api/boards/cy8cproto_063_ble.overlay:
Not really a blocking issue, but header is missing from this file.

sreeramIfx replied:
Will address these in a separate PR.

JarmouniA reviewed (commented):
FYI, having the name of a SoC family/series in a peripheral's compatible does not in anyway prevent its usage in the DT of another family/series (as the underlying hw device is the same).
Also, are the peripherals of non-cat1 SoCs, with which the new compatibles will be potentially used, the same as the cat1 ones at the hw level? Because different HW devices should have different compatibles

sreeramIfx commented:
@maass-hamburg I have updated the migration guide. Please review.

JarmouniA commented on doc/releases/migration-guide-4.4.rst:
```diff
@@ -89,6 +89,41 @@ Ethernet
+
+  Corresponding Kconfig symbols and binding files have also been updated:
+
+  * ``CONFIG_*_INFINEON_CAT1`` → ``CONFIG_*_INFINEON``
```
I believe they should be enumerated one by one.
